lib/travis/api/v3/models/cron.rb

```diff
@@ -0,0 +1,112 @@
+module Travis::API::V3
+  class Models::Cron < Model
+
+    belongs_to :branch
+
+    def start
+      raise ServerError, 'repository does not have a github_id'.freeze unless branch.repository.github_id
+
+      payload = {
+        repository: { id: branch.repository.github_id, owner_name: branch.repository.owner_name, name: branch.repository.name },
+        branch:     branch.name
+      }
+
+      class_name, queue = Query.sidekiq_queue(:build_request)
+      ::Sidekiq::Client.push('queue'.freeze => queue, 'class'.freeze => class_name, 'args'.freeze => [{type: 'cron'.freeze, payload: JSON.dump(payload)}])
+      payload
+    end
+
+    def next_enqueuing
+
+      if (disable_by_build) && (last_non_cron_build_date > last_planned_time)
+        return after_next_planned_time
+      elsif last_cron_build_date >= last_planned_time
+        return next_planned_time
+      else
+        return Time.now
+      end
+    end
+
+    def next_planned_time
```

Comment:
I refactored it, please have a look: 8a6f4066ae45d268a85caff0cd3c93da030c2c23 Do you think there is still a separate object needed?